packages/http-client-csharp/emitter/src/type/input-examples.ts

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+import { InputParameter } from "./input-parameter.js";
+import {
+  InputArrayType,
+  InputDictionaryType,
+  InputLiteralType,
+  InputModelType,
+  InputNullableType,
+  InputPrimitiveType,
+  InputType,
+  InputUnionType,
+} from "./input-type.js";
+import { OperationResponse } from "./operation-response.js";
+
+interface InputExampleBase {
+  kind: string;
+  name: string;
+  description: string;
+  filePath: string;
+  rawExample: any;
+}
+
+export interface InputHttpOperationExample extends InputExampleBase {
+  kind: "http";
+  parameters: InputParameterExampleValue[];
+  responses: Map<number, OperationResponseExample>;
+}
+
+export interface InputParameterExampleValue {
+  parameter: InputParameter;
+  value: InputTypeExampleValue;
+}
+
+export interface OperationResponseExample {
+  response: OperationResponse;
+  // headers: SdkHttpResponseHeaderExample[];
+  bodyValue?: InputTypeExampleValue;
+}
+
+export type InputTypeExampleValue =
+  | InputStringExampleValue
+  | InputNumberExampleValue
+  | InputBooleanExampleValue
+  | InputNullExampleValue
+  | InputAnyExampleValue
+  | InputArrayExampleValue
+  | InputDictionaryExampleValue
+  | InputUnionExampleValue
+  | InputModelExampleValue;
+
+export interface InputExampleTypeValueBase {
+  kind: string;
+  type: InputType;
+  value: unknown;
+}
+export interface InputStringExampleValue extends InputExampleTypeValueBase {
+  kind: "string";
+  type: InputType;
```

Comment:
Shall we narrow-down it  to `InputPrimitiveType` as others ExampleValue type

Reply:
No, other types could have string values, such as literal, enum duration and datetime.
TCGC side has a more detailed type: https://github.com/Azure/typespec-azure/blob/main/packages/typespec-client-generator-core/src/interfaces.ts#L776

Reply:
I did not mimic everything here because it is kind meaningless, because all the things we are doing it here is to convert it and serialize.

Our conversion does not have the magic to take `A | B | C` and gives you `InputA | InputB | InputC`, and it is totally meaningless to write a complicated method to do that, therefore I just simplified it here to InputType.

I think it is not our responsibility to verify all the information we get from TCGC is correct, and it is impossible.